backends-velox/src/test/scala/io/glutenproject/execution/VeloxParquetWriteSuite.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.glutenproject.execution
+
+class VeloxParquetWriteSuite extends WholeStageTransformerSuite {
+  override protected val backend: String = "velox"
+  override protected val resourcePath: String = "/tpch-data-parquet-velox"
+  override protected val fileFormat: String = "parquet"
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    createTPCHNotNullTables()
+  }
+
+  test("test write parquet with compression codec") {
+    // compression codec details see `VeloxParquetDatasource.cc`
+    Seq("snappy", "gzip", "zstd", "none", "uncompressed")
+      .foreach { codec =>
+        val extension = codec match {
+          case "none" | "uncompressed" => ""
+          case "gzip" => "gz"
+          case _ => codec
+        }
+
+        withTempPath { f =>
+          spark.table("customer").write
```

Comment:
it seems there is a bug when I read and write table `lineitem`, can you help check it ? @JkSelf 

```scala
val df = spark.read.format("parquet").load("/gluten/backends-velox/target/scala-2.12/test-classes/tpch-data-parquet-velox/lineitem/part-00000-488993f4-f7ee-41c7-ac6f-b77edaaf70bb-c000.snappy.parquet")
df.write.format("velox").option("compression", "zstd").mode("overwrite").save("/opt/parquet/test")
val res = spark.read.format("parquet").load("/opt/parquet/test")
res.collect
```

```
Caused by: java.lang.RuntimeException: TProtocolException: Invalid data
  at io.glutenproject.vectorized.ColumnarBatchOutIterator.nativeHasNext(Native Method)
  at io.glutenproject.vectorized.ColumnarBatchOutIterator.hasNextInternal(ColumnarBatchOutIterator.java:45)
  at io.glutenproject.vectorized.GeneralOutIterator.hasNext(GeneralOutIterator.java:37)
```

Reply:
I cannot reproduce this issue in unit test. But it core dump in spark shell. I will look into this issue later.

Reply:
Does this issue still exist even with your patch?

Reply:
yes, I think the issue is not about compression codec but an another stuff. I have not found out the root reason though..